src/components/CommentCard.jsx

```diff
@@ -4,30 +4,65 @@ import CardHeader from '@mui/material/CardHeader';
 import Avatar from '@mui/material/Avatar';
 import { red } from '@mui/material/colors';
 import { Typography } from '@mui/material';
+import DeleteOutlinedIcon from '@mui/icons-material/DeleteOutlined';
+import IconButton from '@mui/material/IconButton';
+import { useContext } from 'react';
+import { UserContext } from './contexts';
+import { deleteComment } from '../network/network';
 
-
-
-export default function CommentCard({comment, users}){
+export default function CommentCard({comment, users, comments, setComments}){
     
-    const thisUser = users.filter((user)=>user.username===comment.author)
+    const thisUser = users.find((user)=>user.username===comment.author)
     const date = new Date(comment.created_at) 
-
+    const {user} = useContext(UserContext)
+    
+    
+    const handleDeleteComment = async ()=>{
+        const res = await deleteComment(comment.comment_id)
+        
+        if (await res.status === 204){
```

Comment:
I think this is for optimistic rendering? This should be avoided for operations like delete as it would be hard to undo should an issue occur. Relying on a change to comments in general should suffice.

Reply:
There's no optimistic rendering here. The deleteComment function is part of the network implementation the rest of this function waits for the 204 response before re-rendering the comments in the comments view.